plugins/foreman_templates/4.0/index.md

```diff
@@ -0,0 +1,185 @@
+    <td>N/A</td>
+  </tr>
+  <tr>
+    <td><i>repo</i></td>
+    <td colspan="2">target path, specifies also the protocol, e.g. /tmp/dir or git://example.com, https://example.com, ssh://example.com</td>
+  </tr>
+  <tr>
+    <td><i>verbose</i></td>
+    <td colspan="2">adds extra verbose messages to the action output</td>
+  </tr>
+</table>
+
+Settings above are preconfigured for importing from upstream community-templates repository.
+
+# 4. Usage (features description)
+
+Both importing and exporting is available as rake tasks. Rake tasks are triggered from terminal. If you installed the plugin using foreman-installer, or simply it's package based installation), you should be able to run rake using `foreman-rake` command. If you installed Foreman alternatively, e.g. cloned from git, you should likely use `RAILS_ENV=production bundle exec rake` ran from your Foreman directory. There are other ways to run the action discussed under API chapter.
+ 
+# 4.1 Importing
+
+If you have all attributes configured via Settings you can simply run following command
+
+```sh
+foreman-rake templates:import
+```
+
+This task based on $repo setting either uses git or scans local directory for templates to import. If you want to customize any of setting for a particular run, just specify it as extra argument.
+
+```sh
+foreman-rake templates:import repo=https://github.com/theforeman/community-templates
+```
+
+The example above would clone the repository to some temp directory. If `branch` was specified, it checkouts this git branch. Then if `dirname` was provided, it will "cd" into it. Finally is searches for all files in this directory and subdirectories that has `'.erb` suffix. Then the template is parsed and imported (see below for more details)
+
+```sh
+foreman-rake templates:import repo=/var/lib/my_templates
+```
+
+Compared to previous example, this uses local directory. Therefore not git clone or branch checkout is done. If `dirname` is specified, the task goes into this subdirectory and searches for all files with '.erb' extension. Then the same parsing and importing process follows,
+
+The parsing process reads the content of the file. It expects the content of the template including a metadata. The metadata are an ERB comment on top of the file. It's in yaml format and specifies template name, kind, compatible operating systems, organizations, locations and possibly other. The template gets it's name based on information from metadata. Before the template is saved in database, `prefix` setting is applied to it's name. Then the `filter` regular expression is matched against the prefixed name. If the `filter` matches, templates is saved. If `negate` is set to `true`, the template would be saved only if `filter` regular expression didn't match the template name. If the template with a given name exists in database already, it's content is updated.
```

Comment:
Is it worth linking to an example from community-templates to show a metadata block?